### DISCUSSION TITLE
关于Connection reset by peer (os error 104)警告的疑问

### DISCUSSION THREAD

**Discussion by ioridf1** [shadowsocks/shadowsocks-rust#964] [Q&A]
大神，你们好，
      今天我看到日志中有如下提示，这个IP归属地是荷兰的（不是我的），这是否表示有个ShadowSocks客户端（公网IP为89.248.163.167）已经连上我的服务器了呢？至少密码校验过程已经被攻破了？谢谢！

 WARN  tcp handshake failed. peer: 89.248.163.167:4475, Connection reset by peer (os error 104)

**Comment by zonyitoo** [comment]
This line of log indicates that `recv()` failed with `ECONNRESET`.

https://stackoverflow.com/questions/35093071/tcp-recv-getting-econnreset

There is no evidence about security flaws of your server.

**Reply by ioridf1** [reply]
好的，谢谢大神答复，在您昨天的指点下，我今天看了下https://docs.docker.com/，知道怎么启动容器时运行自定义的命令了，后面准备一步一步慢慢学习如何结合git下来的shadowsocks-rust源码，自行编译出docker。。 我看昨天 v1.15.0-alpha.9已经预发布了，但官方编译的最新docker还没给出来，只能自己动手了，不过这个倒腾研究的过程还是挺开心的:)
